### PR TITLE
chore(backend): format backend code and add style guide

### PR DESCRIPTION
**Pull request by Linchin**
**Description of your changes:**
* Formatted backend code in `backend/src` and `backend/test`
* Added style guide for backend

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,12 +25,15 @@ again.
...
 ### SDK
-See the [SDK-specific Contribution Guidelines](https://github.com/kubeflow/pipelines/blob/master/sdk/README.md) for contribuying to the `kfp` SDK.
+See the [SDK-specific Contribution Guidelines](https://github.com/kubeflow/pipelines/blob/master/sdk/CONTRIBUTING.md) for contribuying to the `kfp` SDK.
 
 ### Frontend
 
 The frontend part of the project uses [prettier](https://prettier.io/) for formatting, read [frontend/README.md#code-style](frontend/README.md#code-style) for more details.
 
+### Backend
+Use [gofmt](https://pkg.go.dev/cmd/gofmt) package to format your .go programs. Read [backend/README.md#code-style](backend/README.md#code-style) for more details.
```
Comment (chensun) [inline CONTRIBUTING.md:35]: nit: programs -> source files?

**Review by chensun — COMMENTED** [review]
Thanks Lingqing! Not a high priority task, but are we going to enforce this via presubmit? 

/lgtm
/cc @gkcalat

**Comment by Linchin** [comment]
@chensun Thanks for the suggestion! Yes I think it would be great to enforce this via presubmit. I will do that in a separate PR.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Linchin) [reply CONTRIBUTING.md:35]: Updated.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -25,12 +25,15 @@ again.
 ## Coding style
 
 ### SDK
-See the [SDK-specific Contribution Guidelines](https://github.com/kubeflow/pipelines/blob/master/sdk/README.md) for contribuying to the `kfp` SDK.
+See the [SDK-specific Contribution Guidelines](https://github.com/kubeflow/pipelines/blob/master/sdk/CONTRIBUTING.md) for contribuying to the `kfp` SDK.
```
Comment (connor-mccarthy) [inline CONTRIBUTING.md:28]: Good catch!

Also, "contributing" -> "contribuying" :)

Comment (connor-mccarthy) [inline CONTRIBUTING.md:28]: Good catch!

Also, "contributing" -> "contribuying" :)

**Review by connor-mccarthy — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (Linchin) [reply CONTRIBUTING.md:28]: I will update this.

**Review by chensun — COMMENTED** [review]
/lgtm
/approve

Thanks @Linchin !

**Comment by Linchin** [comment]
Merged since just some minor changes(typos) after the initial approval and lgtm.

**Pull request merged**